test/nlp/api.jl

```diff
@@ -57,6 +57,9 @@
   Jop = jac_op!(nlp, x, jac_structure(nlp)..., Jv, Jtw)
   @test Jop * v ≈ J(x) * v
   @test Jop' * w ≈ J(x)' * w
+  @test jth_hess(nlp, x, 1) == zeros(2, 2)
+  @test jth_hess_coord(nlp, x, 1) == zeros(2)
+  @test jth_hprod(nlp, x, v, 1) == zeros(2)
```

Comment:
I'm leaving a bit more general so it's easier to copy or modify in the future. Also, it might better with the `for` on line 61 below.
```suggestion
  for j = 1:nlp.meta.ncon
    eⱼ = [i == j ? 1.0 : 0.0 for i = 1:m]
    @test jth_hess(nlp, x, j) == H(x,eⱼ) - H(x)
    @test sparse(hess_structure(nlp)..., jth_hess_coord(nlp, x, j), n, n) == H(x,eⱼ) - H(x)
    @test jth_hprod(nlp, x, v, j) == (H(x,eⱼ) - H(x)) * v
  end
```

Reply:
Cool idea, thanks!